src/Common/src/Internal/Runtime/EETypeOptionalFieldsBuilder.cs

```diff
@@ -93,7 +93,15 @@ public override string ToString()
 
             for (int i = 0; i < (int)EETypeOptionalFieldsElement.Count; i++)
             {
-                sb.Append(_rgFields[i]._value.ToString());
+                if (_rgFields[i]._fieldPresent)
+                {
+                    sb.Append(_rgFields[i]._value.ToString());
```

Comment:
I'll tackle it in a separate PR in that case since the ToString there is pre-existing anyhow.